doc/sphinx/how-to/how-to-enable-graphics-core22-on-a-device.md

```diff
@@ -0,0 +1,455 @@
+0x01 32  0 10 10 10  2  0  0  0 0 0x30335241--     	y  y  y 	win
+0x02 32  0 10 10 10  2 16  0  0 0 0x30335241--     	y  y  y 	win
+0x03 32  0 10 10 10  2 24  0  0 0 0x30335241--     	y  y  y 	win
+0x04 32  0 10 10 10  2 24  8  0 0 0x30335241--     	y  y  y 	win
+0x05 32  0 10 10 10  2  0  0  2 1 0x30335241--     	y  y  y 	win
+… lots more configurations, then other platform details …
+```
+
+If `eglinfo` does **NOT** list a GBM platform, or generates errors then you want to look
+at  [possible complications](#possible-complications). If `eglinfo` does list a GBM platform
+then we can proceed to testing `ubuntu-frame`.
+
+### Test if `ubuntu-frame` works
+First, install `ubuntu-frame`:
+```shell
+sudo snap install ubuntu-frame --devmode
+```
+
+Next, disconnect it from the default graphics interface:
+```shell
+sudo snap disconnect ubuntu-frame:graphics-core22
+```
+
+Then, connect it to your new provider snap:
+```shell
+sudo snap connect ubuntu-frame:graphics-core22 <your-snap>:graphics-core22
+```
+
+Finally, run `ubuntu-frame`:
+```shell
+ubuntu-frame
+```
+
+If everything previously has gone correctly this should result in `ubuntu-frame` coming
+up on the connected outputs. If not, the log messages will hopefully help identify issues.
+
+## How to test the confinement of your *Mir*-based snap
+Once you have your `graphics-core22` provider snap setup, and you are confident that it works
+well, you'll want to make sure that you can run the snap that *depends on* your provider snap
+in a confined mode. Since we already have `ubuntu-frame` setup, we will use it as an example.
+However, you should be able to run these tests on any *Mir*-based snap.
+
+First, let's uninstall `ubuntu-frame` if it is installed:
+
+```shell
+sudo snap remove ubuntu-frame
+```
+
+Next, we can install `ubuntu-frame` without the `--devmode` flag:
+```shell
+sudo snap install ubuntu-frame
+sudo snap disconnect ubuntu-frame:graphics-core22
+```
+
+Finally, we will run `ubuntu-frame` like before:
+```shell
+ubuntu-frame
+```
+
+If everything works, then `ubuntu-frame` is ready. If not, we'll have to troubleshoot.
+
+### Troubleshooting
+When bringing up a confined ubuntu-frame snap on a new board with new drivers there are two
+separate access control mechanisms:
+- AppArmor
+- The devices cgroup
+
+#### AppArmor
+The first is relatively easy to debug. When AppArmor would deny access to a resource, it outputs a nice
+message in `dmesg`:
+
+```
+apparmor="DENIED" operation="open" profile="snap.ubuntu-frame.ubuntu-frame" name="/run/udev/data/c189:1" <other stuff>
+```
+
+You can find the AppArmor profile at `/var/lib/snapd/apparmor/profiles/snap.ubuntu-frame.ubuntu-frame`.
+Make modifications to the profile using your favorite text editor. Finally, apply the change to replace
+the default confinement:
+
+```shell
+sudo apparmor_parser -r /var/lib/snapd/apparmor/profiles/snap.ubuntu-frame.ubuntu-frame
+```
```

Comment:
Probably worth adding a bit about how this profile is regenerated on lots of `snapd` actions, so these modifications are *temporary* and will need to be upstreamed via a bug against snapd to become useful.